### ISSUE TITLE
correct function arg order

### ISSUE THREAD

**Issue by graingert** [jfmengels/eslint-plugin-lodash-fp#49]
When static analysis can trivially work out that the argument order is obviously wrong:

```js
#good
_.map(x => x.y, vals);

#bad
_.map(vals, x => x.y);

#bad
_.map([a, b, c], eg);

#ok
function foo(vals, f) {
  return _.map(vals, f);
}

#bad
function fn(x) {
  return x.y;
}

_.map(vals, fn);
```
